File: frontend/src/models/LegalDescriptionInterpretResponse.ts
import type { AppMode } from './AppMode'
import type { LegalDescriptionBoundary } from './LegalDescriptionBoundary'
import type { LegalDescriptionInterpretation } from './LegalDescriptionInterpretation'
import type { ModeAuditMetadata } from './ModeAuditMetadata'
import type { RetryGuidance } from './RetryGuidance'

export interface LegalDescriptionInterpretResponse {
  mode: Extract<AppMode, 'LegalDescription'>
  schemaVersion: string
  interpretation: LegalDescriptionInterpretation
  boundary?: LegalDescriptionBoundary
  retry?: RetryGuidance
  audit?: ModeAuditMetadata
}

File: backend/src/SteelTree.GeoAcre.Web.Api/Models/LegalDescriptionInterpretResponse.cs
namespace SteelTree.GeoAcre.Web.Api.Models;

public sealed class LegalDescriptionInterpretResponse
{
    public required string Mode { get; set; }

    public required string SchemaVersion { get; set; }

    public required LegalDescriptionInterpretationDto Interpretation { get; set; }

    public LegalDescriptionBoundaryDto? Boundary { get; set; }

    public RetryGuidanceDto? Retry { get; set; }
}
